src/main/java/net/rubygrapefruit/platform/internal/jni/WindowsFileEventFunctions.java

```diff
@@ -16,53 +16,67 @@
 
 package net.rubygrapefruit.platform.internal.jni;
 
-import net.rubygrapefruit.platform.NativeException;
-import net.rubygrapefruit.platform.NativeIntegration;
 import net.rubygrapefruit.platform.file.FileWatcher;
 import net.rubygrapefruit.platform.file.FileWatcherCallback;
 import net.rubygrapefruit.platform.internal.FunctionResult;
 
 import java.util.Collection;
-import java.util.List;
 
-public class WindowsFileEventFunctions implements NativeIntegration {
+public class WindowsFileEventFunctions extends AbstractFileEventFunctions {
 
+    /**
+     * Start watching the given directory hierarchies.
+     *
+     * <h3>Remarks:</h3>
+     *
+     * <ul>
+     *     <li>Changes to any descendants to the given paths are reported.</li>
+     *
+     *     <li>Changes to the given paths themselves are not reported.</li>
+     *
+     *     <li>Changes are reported as <em>canonical</em> paths. This means:
+     *     <ul>
+     *         <li>When watching a path with a different case, the canonical one is used to report changes.</li>
+     *     </ul>
+     *
+     *     <li>Events arrive from a single background thread unique to the {@link FileWatcher}.</li>
+     *
+     *     <li>Removals are reported as a
+     *     {@link net.rubygrapefruit.platform.file.FileWatcherCallback.Type#MODIFIED MODIFIED} and a
+     *     {@link net.rubygrapefruit.platform.file.FileWatcherCallback.Type#REMOVED REMOVED} event.</li>
+     *
+     *     <li>Renames are reported as the source file being
+     *     {@link net.rubygrapefruit.platform.file.FileWatcherCallback.Type#REMOVED REMOVED}.
+     *     The creation of the target file is not reported.</li>
```

Comment:
Ouch. How can we handle that on the Gradle side?

Reply:
Re-read the whole directory every time? :) This is what we do now in any case, so it works.

We can either spend some time to figure out how to make Windows generate an event for the created file, too, or we can post some special event for renames, and... _re-read the whole directory_ on the Gradle side.

I made a note to do these things before we go into production.

Reply:
Which directory? I thought this was about a move to a different directory. Do when only get the `REMOVED` event and not the `CREATED` event in the new location?

Reply:
This is about renaming inside the same watched directory. It’s not move, it’s literally rename. Moving stuff out of the watched dir or into it works well.

Reply:
Perhaps we should clarify these behaviours in the comment, and document that "rename" means "change of file name within the same directory" (which wasn't immediately obvious to me) and perhaps also what happens for things like "move within the watched directory" and "moved into watched directory" and "moved out of watched directory"